### PR TITLE
added dsm

### PR DESCRIPTION
**Pull request by ritwika314**
DSM object added, with shared variable access methods etc.

### CODE REVIEW

File: src/objects/abstract/dsm.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        """
+        return str(self.name)
+
+        # ------------ MEMBER ACCESS METHODS --------------
+
+    @property
+    def name(self) -> str:
+        return self.__name
+
+    @name.setter
```
Comment (hc825b) [inline src/objects/abstract/dsm.py]: Why are setter methods for `__name` and `data_type` necessary?

File: src/objects/dsm_allread.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        :type pid: int
+
+        :param value: value of dsm
+        :type value: int, bool, float, list, object, tuple, None
+
+        :param last_updated: last update
+        :type last_updated: float
+
+        """
+        self.name = name
```
Comment (hc825b) [inline src/objects/dsm_allread.py]: Use `super().__init__(...)` to initialize members of base class.

File: src/objects/dsm_allwrite.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        :type value: int, bool, float, list, object, tuple, None
+
+        :param share_scope: shared variable scope
+        :type share_scope: ShareType
+
+        :param last_updated: last update
+        :type last_updated: float
+
+        """
+        self.name = name
```
Comment (hc825b) [inline src/objects/dsm_allwrite.py]: Use `super().__init__(...)` to initialize members of base class.

File: src/tests/base/test_dsm.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    def test_name(self):
+        self.assertEqual(self.dsm.name, "x", "testing name access")
+
+    def test_dtype(self):
+        self.assertEqual(self.dsm.data_type, int, "testing type access")
+
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (hc825b) [inline src/tests/base/test_dsm.py]: Newline at the end of file (PEP 8).

**Review by hc825b — CHANGES_REQUESTED** [review]
See comments and questions in each file.

Also why `__last_updated` is not in base `dsm` class but in each child class?

File: src/objects/abstract/dsm.py (hunk 1 above)
Comment (ritwika314) [reply src/objects/abstract/dsm.py]: not necessary, can delete.

**Comment by ritwika314** [comment]
> Also why `__last_updated` is not in base `dsm` class but in each child class?

added it as an abstract method now.

**Comment by ritwika314** [comment]
> Also why `__last_updated` is not in base `dsm` class but in each child class?
added it now.

**Review by hc825b — APPROVED** [review]

**Pull request merged**